src/traits/interface.rs

```diff
@@ -0,0 +1,34 @@
+use crate::*;
+
+/// Provides low-level access to a COM interface.
+///
+/// This trait is automatically used by the generated bindings and should not be
+/// used directly.
+pub unsafe trait Interface: Sized + Abi {
+    type Vtable;
+    const IID: Guid;
+
+    unsafe fn vtable(&self) -> &Self::Vtable {
+        self.assume_vtable::<Self>()
+    }
+
+    // TODO: name from assume_init
+    unsafe fn assume_vtable<T: Interface>(&self) -> &T::Vtable {
+        let this: RawPtr = std::mem::transmute_copy(self);
+        &(*(*(this as *mut *mut _) as *mut _))
+    }
+
+    // TODO: picked cast rather than query because there's a WinRT method named query but not one named cast
```

Comment:
What is the todo here?

Reply:
Forgot to push the last commit with more docs. This should be updated.